File: complexity.py
# this uses Radon library. this is a sample from Radon tutorial. 
import os
import radon

# Define the directory path to the microservice
directory_path = '/path/to/microservice'

# Get the list of Python files in the microservice directory
python_files = [os.path.join(directory_path, f) for f in os.listdir(directory_path) if f.endswith('.py')]

# Initialize empty lists for the raw metrics and Cyclomatic Complexity (McCabe)
raw_metrics = []
cc = []

# Loop over each Python file in the microservice directory
for file_path in python_files:
    # Get the raw metrics for the file
    raw_metrics.append(radon.raw.open(file_path).raw_metrics)
    
    # Get the Cyclomatic Complexity (McCabe) for the file
    cc.append(radon.complexity.cc_visit(file_path))

# Print the results
print('Raw Metrics:', raw_metrics)
print('Cyclomatic Complexity:', cc)

File: complexity.js
// this is to measure code complexity of JS microservice - uses escomplex git repo: https://github.com/escomplex/escomplex
// see the repo for more details
const fs = require('fs');
const escomplex = require('escomplex');

// Define the directory path to the microservice
const directoryPath = '/path/to/microservice';

// Get the list of JavaScript files in the microservice directory
const jsFiles = fs.readdirSync(directoryPath).filter(file => file.endsWith('.js'));

// Initialize empty lists for the metrics
const metrics = [];

// Loop over each JavaScript file in the microservice directory
jsFiles.forEach(file => {
  // Read the file and convert to a string
  const fileContent = fs.readFileSync(`${directoryPath}/${file}`, 'utf-8');
  
  // Calculate the complexity metrics for the file
  metrics.push(escomplex.analyse(fileContent));
});

// Aggregate the results
const aggregatedMetrics = metrics.reduce((acc, curr) => {
  acc.aggregate.cyclomatic += curr.aggregate.cyclomatic;
  acc.aggregate.halstead += curr.aggregate.halstead;
  acc.aggregate.sloc += curr.aggregate.sloc;
  acc.methods = [...acc.methods, ...curr.methods];
  acc.functions = [...acc.functions, ...curr.functions];
  return acc;
});

// Print the results
console.log('Aggregated Metrics:', aggregatedMetrics);
